docs/_chapters/230-manifest-annotations.md

```diff
@@ -1,136 +1,286 @@
+results in the manifest header:
 
-	@RequireConfigurer
-	public class Peggy { ... }
+`Bundle-Activator: com.acme.Activator`
 
-Now the same header will be generated.
 
-## Bundle License
+### Macros
 
-The `@BundleLicense` annotation creates entries in the Bundle-License header. The annotation has the following fields:
+You'll note the string `${@class}` used in the above example. String fields in *bundle annotations* are processed through bnd's macro processor. This macro processor provides access to all default and builder macros. *More info on bnd macros can be found in the [macros chapter](/chapters/850-macros.html).* 
 
-* `name` – The name of the license, should preferably the URI to the corresponding [Open Source Initiative][2] page about this license.
-* `description` – A short description of the license.
-* `url` – The URL to further information about the license
+Bnd also provides access to certain key properties of the current processing state.
 
-A number of licenses have gotten their own custom annotation:
+- ***@class*** - gives the fully qualified name of the class where the annotation hierarchy began
+- ***@class-short*** - gives the simple name of the class where the annotation hierarchy began
+- ***@package*** - gives the package of the class where the annotation hierarchy began
+- ***@version*** - gives the package version of the class where the annotation hierarchy began
 
-* `ASL_2_0`
-* `BSD_2_Clause`
-* `BSD_3_Clause`
-* `CDDL_1_0`
-* `CPL_1_0`
-* `EPL_1_0`
-* `GPL_2_0`
-* `GPL_3_0`
-* `LGPL_2_1`
-* `MIT_1_0`
-* `MPL_2_0`
+The `@Header` example above used the macro `${@class}` which lifted the `@class` property holding the class name of the activator into the header to avoid having to duplicate it. This also means that refactoring the activator won't cause the manifest to get out of sync.
 
-See the Javadoc for additional details.
+### Custom Bundle Annotations
 
-### Example
+Certain *bundle annotations* have a second important use. We know that if applied to a type or package *bundle annotations* result in a clause in the manifest. However, many can be used as meta-annotations to a second annotation. The second annotation is considered a ***custom bundle annotation***. The *custom bundle annotation* results in a manifest clause only when applied to a type or package.
 
-	@ASL_2_0
-	public class Peggy { ... }
+This makes it possible to create an annotation for a subsystem. For example, an annotation `@ASL_2_0` that sets the `Bundle-License` header to the Apache Software License version 2.0.
 
-	Bundle-License: \
-		http://www.opensource.org/licenses/apache2.0.php; \
-		description='Apache Software License 2.0';\
-		link='http://www.apache.org/licenses/LICENSE-2.0.html'
+```java
+@BundleLicense(
+    name = "http://www.opensource.org/licenses/apache2.0.php", 
+    link = "http://www.apache.org/licenses/LICENSE-2.0.html", 
+    description = "Apache Software License 2.0")
+@interface ASL_2_0 {}
```

Comment:
We need to fix these annotations for licensing. (1) we should use the names recommended by the OSGi spec and (2) ASL_2 is not a license. It is the Apache License 2. (no S).

Reply:
does this really need to be part of this PR?